rai-sdk/src/main/java/com/relationalai/DefaultPollInterval.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 RelationalAI, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"): you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.relationalai;
+
+import org.awaitility.core.DurationFactory;
+import org.awaitility.pollinterval.PollInterval;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+public class DefaultPollInterval implements PollInterval {
+    private final Double overheadRate;
+    private final Long startTime;
+    private final TimeUnit timeUnit;
+    private final int maxDelay;
+
+    public DefaultPollInterval() {
+        this.overheadRate = 0.1;
+        this.startTime = Instant.now().toEpochMilli();
+        this.timeUnit = TimeUnit.MILLISECONDS;
+        this.maxDelay = 120000; // two minutes
+    }
+    public DefaultPollInterval(Double overheadRate) {
+        this.overheadRate = overheadRate;
+        this.startTime = Instant.now().toEpochMilli();
+        this.timeUnit = TimeUnit.MILLISECONDS;
+        this.maxDelay = 120000; // two minutes
+    }
+
+    public DefaultPollInterval(Long startTime, Double overheadRate, TimeUnit timeUnit, int maxDelay) {
```

Comment:
Nits
* I'd make clear start time should be epoch time in millis by adding a comment
* I'm not sure it makes a lot of sense to expose `timeUnit`. See my other comments on that below. I would instead assume everything is milliseconds & I'd drop time unit. Let's also document that assumption.